src/lib/util/wizard-step.interface.ts

```diff
@@ -46,15 +46,25 @@ export abstract class WizardStep {
   public navigationSymbol: NavigationSymbol = {symbol: ''};
 
   /**
-   * A boolean describing if the wizard step has been completed
+   * An input that controls whether the step is shown as completed when the wizard is presented to the user.
+   *
+   * It is useful to make a step initially completed when it is automatically filled
+   * with some derived/predefined information.
    */
-  public completed = false;
+  // tslint:disable-next-line:no-input-rename
+  @Input('completed')
+  public initiallyCompleted = false;
```

Comment:
> Is there a benefit of using one solution over the other?
No, I don't think so. It is rather a matter of preference.

> In my opinion an awComplete directive leads to a more distinct solution.
Maybe. My idea was that there is already an element that defines a step -- `<aw-wizard-step>`. Why should a property of a step -- whether it is initially completed or not -- be configured with a separate directive rather than a simple `@Input`?

Reply:
> Why should a property of a step -- whether it is initially completed or not -- be configured with a separate directive rather than a simple @Input?

Mainly because of uniformity. Take a look at the [`awOptionalStep`](https://github.com/madoar/angular-archwizard/blob/develop/src/lib/directives/optional-step.directive.ts) directive. It does exactly what you describe. Alternatively `wizardStep.optional` could be set via a simple `@Input`. 

I would like to implement both the `optional` and `completed` behavior using the same "style", because I think it is more intuitive for the user this way and easier for us to maintain. This doesn't mean we need to hold onto the `awOptionalStep` directive, if there are reasons which suggest that setting everything inside the wizard steps is better I'm also fine with removing the `awOptionalStep` directive.

Reply:
Yes, I agree that uniformity is important.

Now I remember why I chose to implement `completed` as an input -- this way it can be easily turned on or off like `[completed]="someCondition"`. With `awCompleted` directive, in order to achieve that we will have to add an input to the directive, but in such case is there much use of the directive itself?

Optional steps (implemented using `awOptionalStep` directive) and completion steps (implemented as a separate `WizardCompletionStepComponent`) rarely need to be conditionally optional / conditionally completion steps. But it is easily to imagine that users will need to mark a step initially completed based on some condition (like can this step be pre-filled with existing data or not).

By the way, the current state of the library lacks uniformity, too -- `awOptionalStep` is a directive, but `WizardCompletionStepComponent` is a separate component.

Reply:
> Now I remember why I chose to implement `completed` as an input -- this way it can be easily turned on or off like `[completed]="someCondition"`. With `awCompleted` directive, in order to achieve that we will have to add an input to the directive, but in such case is there much use of the directive itself?

I agree with you

> Optional steps (implemented using `awOptionalStep` directive) and completion steps (implemented as a separate `WizardCompletionStepComponent`) rarely need to be conditionally optional / conditionally completion steps. But it is easily to imagine that users will need to mark a step initially completed based on some condition (like whether the content of this step can be pre-filled with existing data or not).

I'm not talking about the case a user wants to make step both optional and complete by default.
I'm talking about the case where a user knows that he can make a step optional by adding `awOptionalStep` and therefore "deduces" that he can make a step complete by default by adding `awCompletedStep` to the wizard step. 

> By the way, the current state of the library lacks uniformity, too -- `awOptionalStep` is a directive, but `WizardCompletionStepComponent` is a separate component.

I agree that under a certain viewpoint this looks strange. The main reason for the different implementation approach is that a completion step differs from a normal step by more than a single property/field. A completion step defines its own behavior, which in its initial version differed a lot from a normal step. The idea was also to allow for custom wizard step implementations in distinct components for special behaving steps.

Before the discussion gets out of hand:
I don't mean to say that I'm against settings the "complete by default" flat in the wizard component itself. I just wanted to address the different implementation approaches we can take. This is especially important because I don't think we should change the chosen approach later on, because of backwards compatibility.

Reply:
Thanks for the explanation! To summarize, shall we implement the "initially completed" parameter as `awCompleted` directive with an input to be used like

```html
<!-- makes the step initially completed -->
<aw-wizard-step awCompleted>

<!-- makes the step initially completed if `condition` is `true` -->
<aw-wizard-step [awCompleted]="condition">
```

?

Reply:
I'm fine with that! :)

An additional benefit of splitting such functionality in a new directive is that is makes it easy to divide the tests according to different test scenarios. Tests for the "completed by default" scenarios can then be added to the `spec.ts` file associated with the new directive.

Reply:
Implemented in 5fe547b